### PR TITLE
metrics: remove useless "metric" field in pd.json

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/8381

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

**Comment by rleungx** [comment]
BTW, what is the `metrics` used for?

**Comment by okJiang** [comment]
> BTW, what is the `metrics` used for?

I tried to figure out what it does, but I couldn't find any information about it online.

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -862,7 +862,6 @@
           "interval": "",
           "intervalFactor": 2,
           "legendFormat": "",
-          "metric": "pd_server_tso",
```
Comment (HuSharp) [inline metrics/grafana/pd.json:865]: Does the grafana's official documentation explain what this field does?

Comment (HuSharp) [reply metrics/grafana/pd.json:865]: Oh, got it https://github.com/tikv/pd/pull/8382#issuecomment-2219828864

Comment (okJiang) [reply metrics/grafana/pd.json:865]: Can you give me an approve?

**Review by HuSharp — APPROVED** [review]

**Pull request merged**